contracts/governance/extensions/GovernorSequentialProposalId.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+
+abstract contract GovernorSequentialProposalId is Governor {
+    uint256 private _proposalCount;
+    mapping(uint256 proposalHash => uint256 proposalId) private _proposalIds;
+
+    function getProposalId(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) public view virtual override returns (uint256) {
+        uint256 proposalHash = hashProposal(targets, values, calldatas, descriptionHash);
+        uint256 storedProposalId = _proposalIds[proposalHash];
+        if (storedProposalId == 0) revert GovernorNonexistentProposal(0);
+        return storedProposalId;
+    }
+
+    function _propose(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description,
+        address proposer
+    ) internal virtual override returns (uint256) {
+        uint256 proposalHash = hashProposal(targets, values, calldatas, keccak256(bytes(description)));
+        _proposalIds[proposalHash] = ++_proposalCount;
+
+        return super._propose(targets, values, calldatas, description, proposer);
```

Comment:
This is changing `_propose` in a very significant way: it becomes possible to create multiple proposals with the same contents.

Did you consider this? Is it ok?

I don't have an opinion, but I don't know what other consequences this may have.

Reply:
Just pushed a commit to patch this. It actually would be a critical bug where whenever you repropose the same proposal it gives it a new id and abandons the previous proposal.

Reply:
Wouldn't both old and new proposals just continue to live side by side? In what sense would it be abandoned?

Reply:
Proposals IDs are stored in a mapping of hash to int. So each hash can only be assigned to one proposal id. If you repropose the same proposal it would overwrite previously. As such, any functional calling getProposalId (such as queue and execute) would get the new proposal state and previous votes etc wouldn’t count.